lib/trino-parquet/src/main/java/io/trino/parquet/reader/ShortDecimalColumnReader.java

```diff
@@ -147,4 +163,31 @@ else if (columnDescriptor.getPrimitiveType().getPrimitiveTypeName() == INT64) {
             }
         }
     }
+
+    private void checkBytesFitInShortDecimal(byte[] bytes, int startOffset, Type trinoType)
+    {
+        if (isNegative(bytes, startOffset)) {
```

Comment:
That `startOffset` name is misleading. That's where the check stops, not where it starts.

Also, why do we check `isNegative` on `startOffset` instead of the more obvious `bytes[0]`?

Reply:
You can restructure this method to:
* check if the first value is 0 or -1
* loop to check if all the other values are equal to the first one

I think it will make the implementation shorter and easier to follow.

Reply:
Simplified this method now.
Also renamed startOffset to endOffset here.
I'm checking `bytes[endOffset]` instead of `bytes[0]` so that we show an error when trying to read an encoding where the 8th byte from the end is negative but the remaining upper bytes are 0.